src/runtime/virtcontainers/qemu.go

```diff
@@ -754,6 +758,15 @@ func (q *qemu) createPCIeTopology(qemuConfig *govmmQemu.Config, hypervisorConfig
 	// Deduce the right values for mem-reserve and pref-64-reserve memory regions
 	memSize32bit, memSize64bit := q.arch.getBARsMaxAddressableMemory()
 
+	// Fow now, pcie native hotplug is the only way for Arm to hotadd pci device.
+	if machineType == QemuVirt {
+		virtPcieRootPortNum := len(hypervisorConfig.VhostUserBlkDevices) + defaultVirtPcieRootPortNum
```

Comment:
Why are we adding here an additional root port device if we have supplied let's say two `VhostUserBlkDevices`?

Reply:
I reserve a pcie-root-port for network dev in addition with others.

Reply:
I see its for the net-dev.